File: anagram_detector.rb
def detect_anagram(word1, word2)
  canonical(word1) == canonical(word2)
end

def canonical(word)
  word.downcase.split(//).sort
end

p "Do NOT change any code below this line"
p detect_anagram('angel', 'glean') == true
p detect_anagram('glean', 'angel') == true
p detect_anagram('pants', 'pants') == true
p detect_anagram('CinEmA', 'iceman') == true
p detect_anagram('defgh8', 'g8hefd') == true

File: anagram_detector.exs
defmodule AnagramDetector do
  def canonical(word) do
    String.downcase(word) |> String.split("") |> Enum.sort(&(&1 < &2))
  end

  def detect_anagram(word1, word2) do
    canonical(word1) == canonical(word2)
  end
end

IO.puts AnagramDetector.detect_anagram("angel", "glean") == true
IO.puts AnagramDetector.detect_anagram("glean", "angel") == true
IO.puts AnagramDetector.detect_anagram("pants", "pants") == true
IO.puts AnagramDetector.detect_anagram("abc", "def") == false
IO.puts AnagramDetector.detect_anagram("CinEma", "iceman") == true
IO.puts AnagramDetector.detect_anagram("888grbm", "gr8b8m8") == true
